src/node_crypto.cc

```diff
@@ -5939,6 +5951,14 @@ void GenerateKeyPairEC(const FunctionCallbackInfo<Value>& args) {
   GenerateKeyPair(args, 2, std::move(config));
 }
 
+void GenerateKeyPairEdDSA(const FunctionCallbackInfo<Value>& args) {
+  CHECK(args[0]->IsBoolean());
+  const int id = args[0]->IsTrue() ? EVP_PKEY_ED25519 : EVP_PKEY_ED448;
```

Comment:
mapping true/false to distinct key types works, but seems unnecessarily obscure to me, can't it be some other more obvious values? Even though its a bit more code, the EVP_PKEY_ED25519 and _ED449 constants could be exposed into js, and then passed here directly, then this could be a switch with an `UNREACHABLE()` for any values other than the two supported.